onchain/rollups/contracts/consensus/quorum/Quorum.sol

```diff
@@ -0,0 +1,152 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+pragma solidity ^0.8.8;
+
+import {BitMaps} from "@openzeppelin/contracts/utils/structs/BitMaps.sol";
+
+import {AbstractConsensus} from "../AbstractConsensus.sol";
+import {InputRange} from "../../common/InputRange.sol";
+
+/// @notice A consensus model controlled by a small, immutable set of `n` validators.
+/// @notice You can know the value of `n` by calling the `numOfValidators` function.
+/// @notice Upon construction, each validator is assigned a unique number between 1 and `n`.
+/// These numbers are used internally instead of addresses for optimization reasons.
+/// @notice You can list the validators in the quorum by calling the `validatorById`
+/// function for each ID from 1 to `n`.
+contract Quorum is AbstractConsensus {
+    using BitMaps for BitMaps.BitMap;
+
+    /// @notice The total number of validators.
+    /// @notice See the `numOfValidators` function.
+    uint256 private immutable _numOfValidators;
+
+    /// @notice Validator IDs indexed by address.
+    /// @notice See the `validatorId` function.
+    /// @dev Non-validators are assigned to ID zero.
+    /// @dev Validators have IDs greater than zero.
+    mapping(address => uint256) private _validatorId;
+
+    /// @notice Validator addresses indexed by ID.
+    /// @notice See the `validatorById` function.
+    /// @dev Invalid IDs are assigned to address zero.
```

Comment:
I mean, I understand that we're not doing assignments as in Solidity statements, but I used the word "assign" as in English you'd say "I was assigned this task" or "He was assigned a unique number".

Reply:
Ok, maybe saying that "invalid IDs are assigned to address zero" may be misunderstood as "address zero is assigned to invalid IDs", while this might not be true! Address zero can be, theoretically, a validator and have a valid ID. So, saying that invalid IDs map to address zero is more correct.